Sources/Player/OfflineEngine/Internal/DownloadTask.swift

```diff
@@ -127,6 +123,54 @@ final class DownloadTask {
 }
 
 private extension DownloadTask {
+	func calculateSize() -> Int {
+		guard let playbackInfo, let localAssetUrl else {
+			return 0
+		}
+
+		if playbackInfo.mediaType == MediaTypes.HLS ||
+			(playbackInfo.productType == .VIDEO && playbackInfo.mediaType == MediaTypes.EMU)
+		{
+			return calculateHLSStreamSize(for: localAssetUrl)
+		} else {
+			return calculateProgressiveFileSize(for: localAssetUrl)
+		}
+	}
+
+	func calculateHLSStreamSize(for path: URL) -> Int {
+		var totalSize: Int = 0
+
+		if let enumerator = PlayerWorld.fileManagerClient.enumerator(
+			at: path,
+			includingPropertiesForKeys: [URLResourceKey.fileSizeKey]
+		) {
+			for case let fileURL as URL in enumerator {
+				do {
+					let fileAttributes = try fileURL.resourceValues(forKeys: [URLResourceKey.fileSizeKey])
+					if let fileSize = fileAttributes.fileSize {
+						totalSize += Int(fileSize)
+					}
+				} catch {
+					PlayerWorld.logger?.log(loggable: PlayerLoggable.failedToCalculateSizeForHLSDownload(error: error))
+				}
```

Comment:
[suggestion] Minor, I think if we use `reduce` we would reduce the indentation by 1, readability would be mostly same I guess.

Reply:
valid suggestion. I have a few stacked PRs so I think I won't bother as the difference is minimal. 
But I appreciate it!